### PR TITLE
Decode UintOptRle with floating number for negative zero

### PR DESCRIPTION
**Pull request by namse**
Related issue: https://github.com/y-crdt/y-crdt/issues/188

I found that lib0 has wrong port about negative value checking for negative zero in integer type. 
I changed it to use floating number for `UIntOptRleDecoder`.

I'm not sure that we have to cover all cases or just `UIntOptRleDecoder`.

### CODE REVIEW

File: lib0/src/number.rs [hunk 1]
```diff
@@ -284,3 +284,67 @@ fn read_var_i64<R: Read>(reader: &mut R) -> Result<i64, Error> {
         }
     }
 }
+
+impl VarInt for f64 {
```
Comment (namse) [inline lib0/src/number.rs:288]: It would sounds strange that Var`Int` but impl for `f`64.

**Comment by Horusiath** [comment]
First thank you for your investigation. However I don't think it's a good idea to indtroduce `VarInt` trait for `f64`, simply because the issue is not about floating point numbers. The source of an issue exists because `VarInt::read` doesn't include sign information in case of `-0`. It seems that better idea would be to have a dedicated method like `fn read_var_with_sign() -> Result<(i64,i8), Error>` which will return sign information together with a decoded value.

**Comment by namse** [comment]
Sure, I think so. here is my suggestion.

```rust
struct Signed<T> {
  value: T,
  sign: bool,
}

impl<T> Signed<T> {
  pub fn is_positive(&self) -> bool {
    self.sign
  }
  pub fn is_negative(&self) -> bool {
    -self.sign
  }
}

fn read_var_signed() -> Result<Signed, Error> {
  ...
}
```

@Horusiath 
I haven't checked all the code but just made this PR, so I'm a little worried that if I do this it will be insufficient work.
Will you do the sign work yourself? If so, I'll close the PR.
Please let me know if you think it would be better for me to do it.

**Comment by Horusiath** [comment]
I like you suggestion. It expresses the intent in clean way. Feel free to contribute with it. Otherwise I'll probably work on it on the weekend/next week.

**Pull request closed without merging**